pkg/runtime/wfengine/activity.go

```diff
@@ -280,8 +291,30 @@ func (a *activityActor) saveActivityState(ctx context.Context, actorID string, s
 	return nil
 }
 
+func (a *activityActor) purgeActivityState(ctx context.Context, actorID string, state activityState) error {
+	req := actors.TransactionalRequest{
+		ActorType: ActivityActorType,
+		ActorID:   actorID,
+		Operations: []actors.TransactionalOperation{{
+			Operation: actors.Delete,
+			Request: actors.TransactionalDelete{
+				Key: getActivityInvocationKey(state.Generation),
+			},
+		}},
+	}
+	if err := a.actorRuntime.TransactionalStateOperation(ctx, &req); err != nil {
+		return fmt.Errorf("failed to save activity state: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to delete activity state: %w", err)
```

Also, let's add the key in this error message to make debugging easier/possible.